File: api/src/dto/auth.dto.ts
import { UserDTO } from './user.dto.js';
import { IdDTO, TimestampDTO } from './common.dto.js';
// ----------------------
// Auth
// ----------------------

export interface LoginDTO {
  email: string;
  password: string;
}

export interface RegisterDTO {
  email: string;
  password: string;
  displayName?: string;
}

export interface RefreshDTO {
  refreshToken: string;
}

export interface LogoutDTO {
  refreshToken: string;
}

export interface OAuthLoginDTO {
  idToken: string;
}

export interface ChangePasswordDTO {
  currentPassword: string;
  newPassword: string;
}

export interface ChangeEmailDTO {
  currentPassword: string;
  newEmail: string;
}

/**
 * Tokens
 */
export interface AuthTokensDTO {
  accessToken: string;
  refreshToken: string;
}

/**
 * Login / OAuth response
 */
export interface AuthResponseDTO {
  tokens: AuthTokensDTO;
  user: UserDTO;
}

/**
 * Refresh response
 */
export type RefreshResponseDTO = AuthTokensDTO;

/**
 * Auth session (optional exposure / admin)
 */
export interface AuthSessionDTO extends IdDTO, TimestampDTO {
  userId: string;
  device?: string | null;
  ip?: string | null;
  userAgent?: string | null;
  expiresAt: string;
  lastUsedAt: string;
}

File: clients/apple/Packages/AudiobookCore/Sources/DTO/AuthDTO.swift
import Foundation

public struct LoginRequestDTO: Encodable {
    public let email: String
    public let password: String

    public init(email: String, password: String) {
        self.email = email
        self.password = password
    }
}

public struct RegisterRequestDTO: Encodable {
    public let email: String
    public let password: String
    public let displayName: String?
    public let preferredLocale: String?

    public init(email: String, password: String, displayName: String?, preferredLocale: String?) {
        self.email = email
        self.password = password
        self.displayName = displayName
        self.preferredLocale = preferredLocale
    }
}

public struct ChangePasswordRequestDTO: Encodable {
    public let currentPassword: String
    public let newPassword: String

    public init(currentPassword: String, newPassword: String) {
        self.currentPassword = currentPassword
        self.newPassword = newPassword
    }
}

public struct ChangeEmailRequestDTO: Encodable {
    public let currentPassword: String
    public let newEmail: String

    public init(currentPassword: String, newEmail: String) {
        self.currentPassword = currentPassword
        self.newEmail = newEmail
    }
}

public struct RefreshRequestDTO: Encodable {
    public let refreshToken: String

    public init(refreshToken: String) {
        self.refreshToken = refreshToken
    }
}

public struct LogoutRequestDTO: Encodable {
    public let refreshToken: String

    public init(refreshToken: String) {
        self.refreshToken = refreshToken
    }
}

public enum OAuthProviderDTO: String {
    case google
    case apple
}

public struct OAuthLoginRequestDTO: Encodable {
    public let idToken: String

    public init(idToken: String) {
        self.idToken = idToken
    }
}

public struct AuthTokensDTO: Decodable {
    public let accessToken: String
    public let refreshToken: String
}

public struct AuthUserDTO: Decodable {
    public let id: String
}

public struct AuthResponseDTO: Decodable {
    public let tokens: AuthTokensDTO
    public let user: AuthUserDTO
}

public struct RefreshResponseDTO: Decodable {
    public let accessToken: String
    public let refreshToken: String
}
